docs/website/docs/walkthroughs/create-a-pipeline.md

```diff
@@ -34,106 +46,115 @@ Install the dependencies necessary for DuckDB:
 pip install -r requirements.txt
 ```
 
-## 2. Add WeatherAPI.com API credentials
-
-You will need to [sign up for the WeatherAPI.com API](https://www.weatherapi.com/signup.aspx).
+## 2. Obtain and add API credentials from GitHub
 
-Once you do this, you should see your `API Key` at the top of your
-[user page](https://www.weatherapi.com/my/).
+You will need to [sign in](https://github.com/login) to your github account and create your access token via [Personal access tokens page](https://github.com/settings/tokens).
 
-Copy the value of the API key into `.dlt/secrets.toml`:
+Copy your new access token over to `.dlt/secrets.toml`:
 
 ```toml
 [sources]
 api_secret_key = '<api key value>'
 ```
-The **secret name** corresponds to the **argument name** in the source function. Below `api_secret_key` [will get its value](../general-usage/credentials/configuration.md#general-usage-and-an-example) from `secrets.toml` when `weatherapi_source()` is called.
+
+This token will be used by `githubapi_source()` to authenticate requests.
+
+The **secret name** corresponds to the **argument name** in the source function.
+Below `api_secret_key` [will get its value](../general-usage/credentials/configuration.md#general-usage-and-an-example) from `secrets.toml` when `githubapi_source()` is called.
+
 ```py
 @dlt.source
-def weatherapi_source(api_secret_key=dlt.secrets.value):
-  ...
+def githubapi_source(api_secret_key: str = dlt.secrets.value):
+    return githubapi_resource(api_secret_key=api_secret_key)
 ```
 
-Run the `weatherapi.py` pipeline script to test that authentication headers look fine:
+Run the `githubapi.py` pipeline script to test that authentication headers look fine:
 
 ```sh
-python3 weatherapi.py
+python3 githubapi.py
 ```
 
 Your API key should be printed out to stdout along with some test data.
 
-## 3. Request data from the WeatherAPI.com API
+## 3. Request project issues from then GitHub API
 
-Replace the definition of the `weatherapi_resource` function definition in the `weatherapi.py`
-pipeline script with a call to the WeatherAPI.com API:
 
-```py
-@dlt.resource(write_disposition="append")
-def weatherapi_resource(api_secret_key=dlt.secrets.value):
-    url = "https://api.weatherapi.com/v1/current.json"
-    params = {
-        "q": "NYC",
-        "key": api_secret_key
-    }
-    response = requests.get(url, params=params)
-    response.raise_for_status()
-    yield response.json()
-```
+>[!NOTE]
+> We will use dlt as an example project https://github.com/dlt-hub/dlt, feel free to replace it with your own repository.
 
-Run the `weatherapi.py` pipeline script to test that the API call works:
+Modify `githubapi_resource` in `githubapi.py` to request issues data from your GitHub project's API:
```

Comment:
I'd just add one sentence: explain how we picked BearerTokenAuth and HeaderLinkPaginator: by referring to a particular docs page of Github. otherwise this looks like magic